core/Tracker/GoalManager.php

```diff
@@ -671,16 +671,36 @@ protected function recordStandardGoals(VisitProperties $visitProperties, Request
             }
 
             // If multiple Goal conversions per visit, set a cache buster
-            $conversion['buster'] = $convertedGoal['allow_multiple'] == 0
-                ? '0'
-                : $visitProperties->getProperty('visit_last_action_time');
+            if ($convertedGoal['allow_multiple'] == 0) {
+                $conversion['buster'] = 0;
+            } else {
+                $lastActionTime = $visitProperties->getProperty('visit_last_action_time');
+                if (empty($lastActionTime)) {
+                    $conversion['buster'] = $this->makeRandomMySqlUnsignedInt(10);
+                } else {
+                    $conversion['buster'] = $this->makeRandomMySqlUnsignedInt(2) . Common::mb_substr($visitProperties->getProperty('visit_last_action_time'), 2);
+                }
+            }
 
             $conversionDimensions = ConversionDimension::getAllDimensions();
             $conversion = $this->triggerHookOnDimensions($request, $conversionDimensions, 'onGoalConversion', $visitor, $action, $conversion);
 
             $this->insertNewConversion($conversion, $visitProperties->getProperties(), $request, $action, $convertedGoal);
         }
     }
+    
+    private function makeRandomMySqlUnsignedInt($length)
+    {
+        // mysql int unsgined max value is 4294967295 so we cannot start with a 5 or higher
+        $randomInt = Common::getRandomString(1, '1234');
+        if ($length >= 2) {
+            $randomInt .= Common::getRandomString(1, '012');
+        }
+        if ($length > 2) {
+            $randomInt .= Common::getRandomString($length - 2, '0123456789');
+        }
```

Comment:
Actually even that could generate a too high number, even though the range is now much smaller between 4294967295 and 4299999999. Maybe it would be simpler to restrict the number to be between 1000000000 and 3999999999, e.g. disallow the 4 as first char?

Reply:
sure, makes totally sense.

Reply:
changed it.